### PR TITLE
NCI Gadi HPC config 

### PR DESCRIPTION
**Pull request by georgiesamaha**
---
name: NCI Gadi
about: A new cluster config
---

Please follow these steps before submitting your PR:

- [x] Your PR targets the `master` branch
- [x] You've included links to relevant issues, if any
- [ ] Requested review from @nf-core/maintainers and/or #request-review on slack

Steps for adding a new config profile:

- [x] Add your custom config file to the `conf/` directory
- [x] Add your documentation file to the `docs/` directory
- [x] Add your custom profile to the `nfcore_custom.config` file in the top-level directory
- [x] Add your custom profile to the `README.md` file in the top-level directory
- [x] Add your profile name to the `profile:` scope in `.github/workflows/main.yml`

### CODE REVIEW

File: docs/nci_gadi.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Before running the pipeline you will need to load Nextflow and Singularity, both of which are globally installed modules on Gadi. You can do this by running the commands below:
+
+```bash
+module purge
+module load nextflow singularity
+```
+
+### Execution command
+
+```
```
Comment (mashehu) [inline docs/nci_gadi.md:24]:
```suggestion
```bash
```
Gives nice code highlighting and the copy code button.

File: docs/nci_gadi.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+- Tasks requesting **more than 128 Gb and less than 190 Gb** will be submitted to the normal queue
+- Tasks requesting **more than 190 Gb and less than 1020 Gb** will be submitted to the hugemembw queue
+
+See the NCI Gadi [queue limit documentation](https://opus.nci.org.au/display/Help/Queue+Limits) for details on charge rates for each queue.
+
+### Project accounting
+
+This config uses the PBS environmental variable `$PROJECT` to assign a project code to all task job submissions for billing purposes. If you are a member of multiple Gadi projects, you should confirm which project will be charged for your pipeline execution. You can do this using:
+
+```
```
Comment (mashehu) [inline docs/nci_gadi.md:50]:
```suggestion
```bash
```

File: docs/nci_gadi.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+echo $PROJECT
+```
+
+The version of Nextflow installed on Gadi has been modified to make it easier to specify resource options for jobs submitted to the cluster. See NCI's [Gadi user guide](https://opus.nci.org.au/display/DAE/Nextflow) for more details. You can manually override the $PROJECT specification by editing your local copy of the `nci_gadi.config` and replacing $PROJECT with your project code. For example:
+
+```
```
Comment (mashehu) [inline docs/nci_gadi.md:56]:
```suggestion
```nextflow
```

File: docs/nci_gadi.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    storage = 'scratch/aa00+gdata/aa00'
+    ...
+}
+```
+
+## Resource usage
+
+The NCI Gadi config summarises resource usage in a custom trace file that will be saved to your execution directory. However, for accounting or resource benchmarking purposes you may need to collect per-task SU charges. Upon workflow completion, you can run the Sydney Informatics Hub's [gadi_nfcore_report.sh](https://github.com/Sydney-Informatics-Hub/HPC_usage_reports/blob/master/Scripts/gadi_nfcore_report.sh) script in your workflow execution directory with:
+
+```
```
Comment (mashehu) [inline docs/nci_gadi.md:69]:
```suggestion
```bash
```

File: docs/nci_gadi.md (hunk 1 above)
Comment (georgiesamaha) [reply docs/nci_gadi.md]: Thanks @mashehu have incorporated this and all suggestions below in my fixes :)

**Review by jfy133 — COMMENTED** [review]
Sorry for lack of suggestions, doing from my phone:

1. If you only have single file, you don't need the `profile{}` block, you just can just put all the current contents into the file itself
2. It might be unclear what is means by 'download' a local copy. If your head/login node has internet access, you can just get the users to `nextflow pull` there to get the pipeline code (but this won't pull the containers!) As you already suggest people run the `run` command in a screen session, I would suggest saying they always do that. Alternatively you can use users to use `nf-core download` instead (docs:  https://nf-co.re/docs/usage/offline)

**Comment by georgiesamaha** [comment]
Thanks very much @jfy133! I have incorporated your feedback and made adjustments :)

File: conf/nci_gadi.config [hunk 5]
```diff
@@ -0,0 +1,47 @@
+// NCI Gadi nf-core configuration profile
+params {
+    config_profile_description = 'NCI Gadi HPC profile provided by nf-core/configs'
+    config_profile_contact = 'Georgie Samaha (@georgiesamaha), Matthew Downton (@mattdton)'
+    config_profile_url = 'https://opus.nci.org.au/display/Help/Gadi+User+Guide'
+    project = System.getenv("PROJECT")
+}
+
+
+nci_gadi {
```
Comment (jfy133) [inline conf/nci_gadi.config:10]:
```suggestion
```

File: conf/nci_gadi.config [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+    // Write custom trace file with outputs required for SU calculation
+    def trace_timestamp = new java.util.Date().format('yyyy-MM-dd_HH-mm-ss')
+    trace {
+        enabled = true
+        overwrite = false
+        file = "./gadi-nf-core-trace-${trace_timestamp}.txt"
+        fields = 'name,status,exit,duration,realtime,cpus,%cpu,memory,%mem,rss'
+    }
+}
```
Comment (jfy133) [inline conf/nci_gadi.config:47]:
```suggestion
```

**Review by jfy133 — APPROVED** [review]
Last change but otherwise lgtk now :) (ofc, test if possible)

**Comment by georgiesamaha** [comment]
Thanks very much for the thorough review @jfy133, you taught me some new things!

**Pull request merged**